### PR TITLE
Remove EverCrypt.CTR from the tree

### PR DESCRIPTION
**Pull request by msprotz**
CTR is a deprecated API, this module is unverified, and currently has no known users.

The HACL maintainers have decided to relocate EverCrypt.CTR outside of the hacl-star repository. This PR removes it from hacl-star/hacl-star

CC @karthikbhargavan @polubelova @R1kM

See https://github.com/project-everest/everquic-crypto/pull/5

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -693,7 +693,7 @@ TARGETCONFIG_FLAGS = -add-include '"evercrypt_targetconfig.h"'
 # corresponding feature, one of these variables needs to be overridden.
 E_HASH_BUNDLE=-bundle EverCrypt.Hash+EverCrypt.Hash.Incremental=[rename=EverCrypt_Hash]
 MERKLE_BUNDLE=-bundle 'MerkleTree+MerkleTree.EverCrypt+MerkleTree.Low+MerkleTree.Low.Serialization+MerkleTree.Low.Hashfunctions=MerkleTree.*[rename=MerkleTree]'
-CTR_BUNDLE=-bundle EverCrypt.CTR=EverCrypt.CTR.*
+CTR_BUNDLE=-bundle EverCrypt.CTR.*
```
Comment (polubelova) [inline Makefile:696]: why do you need to keep this bundle?

**Review by polubelova — APPROVED** [review]
Thanks a lot!

File: Makefile (hunk 1 above)
Comment (msprotz) [reply Makefile:696]: yes, there is still EverCrypt.CTR.Keys, which contains the key-expansion wrappers shared between AEAD and CTR... happy to do a followup to rename this module?

**Comment by msprotz** [comment]
Marina (thanks!) points out that this closes both #597 and #392, and also improves #649

**Comment by msprotz** [comment]
Screen Shot 2022-10-20 at 7 34 34 PM

**Review by karthikbhargavan — APPROVED** [review]
Unverified code like this is scary, we need to get rid of this and anything else we find.

**Pull request merged**